packages/pocket-ic/tests/tests.rs

```diff
@@ -3587,27 +3587,51 @@ fn payload_too_large() {
     );
     for url in [instances_url, gateway_url] {
         let client = reqwest::blocking::Client::new();
-        let resp = client
-            .post(url)
-            .header(reqwest::header::CONTENT_TYPE, "application/cbor")
-            .body(vec![42; 5 * 1024 * 1024])
-            .send()
-            .unwrap();
-
-        assert_eq!(resp.status(), StatusCode::PAYLOAD_TOO_LARGE);
-        let bytes = String::from_utf8(resp.bytes().unwrap().to_vec()).unwrap();
-        assert!(bytes.contains("error: payload_too_large\ndetails: Payload is too large: maximum body size is 4194304 bytes."));
+        while let Err(err) = send_too_large_body(
+            &client,
+            &url,
+            StatusCode::PAYLOAD_TOO_LARGE,
+            "error: payload_too_large\ndetails: Payload is too large: maximum body size is 4194304 bytes.",
+        ) {
+            println!("{err}");
+            std::thread::sleep(Duration::from_millis(100));
+        }
```

Comment:
To be more robust, wouldn't we want to have a maximum number of retries? If not, then in theory this would end up being an infinite loop and the test would still timeout, or am I missing something?